tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -23,25 +23,90 @@
 
 namespace clang {
 namespace {
-class SPIRVEmitter : public ASTConsumer,
-                     public RecursiveASTVisitor<SPIRVEmitter> {
+
+class SPIRVEmitter : public ASTConsumer {
 public:
   explicit SPIRVEmitter(raw_ostream *Out)
-      : OutStream(*Out), TheContext(), Builder(&TheContext) {}
+      : OutStream(*Out), TheContext(), TheBuilder(&TheContext) {}
 
   void HandleTranslationUnit(ASTContext &Context) override {
-    Builder.beginModule();
-    Builder.endModule();
-    std::vector<uint32_t> M = Builder.takeModule();
+    TheBuilder.requireCapability(spv::Capability::Shader);
```

Comment:
Do compute kernels also need `OpCapability Shader` ?
They probably need `OpCapability Kernel`, in which case, we may want to take the shader type (graphic vs compute) as a command line option.

https://msdn.microsoft.com/en-us/library/windows/desktop/ff476330(v=vs.85).aspx 
has a sample HLSL compute shader.